internal/services/search/search_service_resource.go

```diff
@@ -434,6 +452,17 @@ func resourceSearchServiceUpdate(d *pluginsdk.ResourceData, meta interface{}) er
 		}
 	}
 
+	if d.HasChange("semantic_search_sku") {
+		semanticSearchSku := d.Get("semantic_search_sku").(string)
+
+		// NOTE: Semantic Search SKU cannot be set if the SKU is 'free'
+		if pointer.From(model.Sku.Name) == services.SkuNameFree && semanticSearchSku != string(services.SearchSemanticSearchDisabled) {
+			return fmt.Errorf("'semantic_search_sku' can only be set to %q if the 'sku' field is set to %q, got %q", string(services.SearchSemanticSearchDisabled), string(services.SkuNameFree), semanticSearchSku)
+		}
+
+		model.Properties.SemanticSearch = pointer.To(services.SearchSemanticSearch(semanticSearchSku))
```

Comment:
Which means this can become:

```suggestion
		semanticSearchSku := services.SearchSemanticSearchDisabled
		if v := d.Get("semantic_search_sku").(string); v != "" {
			semanticSearchSku = services.SearchSemanticSearch(v)
		}

		// NOTE: Semantic Search SKU cannot be set if the SKU is 'free'
		if pointer.From(model.Sku.Name) == services.SkuNameFree && semanticSearchSku != services.SearchSemanticSearchDisabled {
			return fmt.Errorf("`semantic_search_sku` can only be specified when `sku` is not set to %q string(services.SkuNameFree)
		}

		model.Properties.SemanticSearch = pointer.To(semanticSearchSku)
```

To allow this property to be removed/marked as `null` in the configuration to remove this field

Reply:
Fixed.